src/custom_modules/DAO.ts

```diff
@@ -69,24 +87,50 @@ export const DAO = {
     cb();
   },
 
-  query: async function <T>(query, params = undefined, retries = 0) {
+  query: async function <T = any>(
+    query,
+    params = undefined,
+    retries = 0,
+  ): Promise<
+    [
+      T extends Array<Record<string, any>>
+        ? RawPrismaTypes<T>
```

Comment:
I'm not sure if I should use the `RawPrismaType` generic here, or if the consumer should use it instead, e.g

```typescript
DAO.query<RawPrismaType<Payment>>(/* ... */)
```

Reply:
Hm, not sure I'm not following what determines if a list is returned or not. I would have thought a list was always returned. Is that some magic inside `dbPool.query` trying to understand if the query will only yield 1 row?

I think it's generally nicer to have that explicit (`queryOne`/`queryMany` or similar). Would simplify this return type a lot also. And then it wouldn't be tempting to have the consumer specify the type as we would always know right?

But maybe that would be fighting mysql2 too much?

Reply:
Didn't think about this, it would seem very intuitive that it always returns an array. However, it seems like for some queries `mysql2` can return an object, for example on executing an `INSERT` query.

Instead of `RawPrismaType` however, we could use a more generic name, since we know any SQL query will not produce types like Date etc..